tradeexecutor/state/visualisation.py

```diff
@@ -358,10 +359,30 @@ class Visualisation:
     #:
     plots: Dict[str, Plot] = field(default_factory=dict)
 
+    #: Trading pair IDs to visualise instead of the default trading pairs from 
+    #: strategy universe.
+    pair_ids: list[PrimaryKey] = field(default_factory=list)
+
     def __repr__(self) -> str:
         plot_count = len(self.plots)
         point_count = sum([len(p.points) for p in self.plots.values()])
         return f"<Visualisation with {plot_count} plots and {point_count} data points>"
+    
+    def set_visualised_pairs(self, pairs: list[TradingPairIdentifier | DEXPair]) -> None:
+        """Set the trading pair for this plot.
+
+        :param pairs:
+            Use these trading pairs to plot candle charts instead of the default
+            trading pairs from strategy universe.
+        """
+        assert isinstance(pairs, list)
+        for pair in pairs:
+            if isinstance(pair, DEXPair):
```

Comment:
DEXPair should never happen here, as new strategy functions always return TradingPairIdentifier.

If they do not, the pair can be converted earlier.

Reply:
Fixed